### ISSUE TITLE
source ~/.zshrc required on each restart of terminal

### ISSUE THREAD

**Issue by Tset-Noitamotua** [caiogondim/bullet-train-oh-my-zsh-theme#193]
I have to `source` my `.zshrc` each time I open a new terminal window. How can I fix it?

![image](https://cloud.githubusercontent.com/assets/5635462/21958852/d94b21b2-dab7-11e6-9cc1-011fc56a795e.png)

**Comment by dawikur** [comment]
Is this question related to `bullet-trian-j-my-zsh-theme` project or rather to `zsh` itself?

**Comment by sfarzy** [comment]
I think this is not related to the theme. This is most likely due to virtualenwrapper. These threads might help. [1](https://stackoverflow.com/questions/19549824/terminal-issue-with-virtualenvwrapper-after-mavericks-upgrade) , [2](https://github.com/sashahart/vex/issues/13)

**Comment by Tset-Noitamotua** [comment]
Oh, sorry for missleading you by the screenshot. Please ignore the Python error messages (thats another issue with my multiple Python versions settup I guess). 

Instead Please take particular note at the icons (for 'dirty', 'ahead' and 'untracked' states) in the prompt.

The first prompt in the screenshot shows how it looked in last terminal session (where I already sourced `.zshrc`). When I start a new terminal I see a prompt like the second one  in the screenshot - it has the default icons that are defined in [bullet-train.zsh-theme](https://github.com/caiogondim/bullet-train-oh-my-zsh-theme/blob/master/bullet-train.zsh-theme).

I think it is related to `bullet-train-oh-my-zsh-theme`.

**Comment by dawikur** [comment]
@Tset-Noitamotua How is it related? Sorry, for me it looks like some zsh configuration issue.

**Comment by Tset-Noitamotua** [comment]
I have not configured much on zsh itself beside of this theme. Maybe I misconfigured something in  my [.zshrc](https://gist.github.com/Tset-Noitamotua/8354a0925f0f0d602b931909cff25db5#file-zshrc) which overides the defaults of `bullet-train.zsh-theme`.? Maybe I need to place my theme customization before line [56](https://gist.github.com/Tset-Noitamotua/8354a0925f0f0d602b931909cff25db5#file-zshrc-L56)?

**Comment by dawikur** [comment]
Yes, first you need to set all settings and than load theme. This is align with what you see:
When `zsh` start it loads `zshrc` - first loads theme, than customize it (but because it was loaded *before* customization it runs with default settings). Then you are sourcing again `zshrc`, which loads again theme, but because now your customization is set (by first load of `zshrc`) it looks as you want it to.

**Comment by Tset-Noitamotua** [comment]
Thanks for looking into it. I moved my customization code above line `SH_THEME="bullet-train"` and I don't need to source after each restart any more which is big progress for me :)

Unfortunately now this part (battery + cafe cup) does not have any effect any more :(
```zsh
prompt_custom() {
  if $(git rev-parse --is-inside-work-tree >/dev/null 2>&1); then
    if [[ -n $(git status --porcelain --ignore-submodules) ]]; then
      # this just avoids that ,, icons are to close to the border
      prompt_segment 197 000 "%F{255}%F{025}"
    fi
  fi
  prompt_segment 230 236 "%F{236}  %F{223}  "
}
```

It does not appear any more

![image](https://cloud.githubusercontent.com/assets/5635462/22004900/94d1ef62-dc5f-11e6-81e7-fd42d5f415d3.png)


Any idea on that?


## my custom code
```zsh
BULLETTRAIN_PROMPT_ORDER=(
time
status
virtualenv
context
dir
git
cmd_exec_time
custom
)

prompt_custom() {
  if $(git rev-parse --is-inside-work-tree >/dev/null 2>&1); then
    if [[ -n $(git status --porcelain --ignore-submodules) ]]; then
      # this just avoids that ,, icons are to close to the border
      prompt_segment 197 000 "%F{255}%F{025}"
    fi
  fi
  prompt_segment 230 236 "%F{236}  %F{223}  "
}

BULLETTRAIN_VIRTUALENV_PREFIX=""
BULLETTRAIN_GIT_PREFIX="%F{230}   %F{230}" 
BULLETTRAIN_GIT_SUFFIX="%F{196}"
BULLETTRAIN_PROMPT_CHAR=

#BULLETTRAIN_TIME_BG=025           # blue
#BULLETTRAIN_TIME_FG=032           # neongruen 118

BULLETTRAIN_TIME_BG=236           # grey dark
BULLETTRAIN_TIME_FG=230           # creme

BULLETTRAIN_VIRTUALENV_BG=039     # python blue
BULLETTRAIN_VIRTUALENV_FG=227     # python yellow

BULLETTRAIN_DIR_BG=238            # blue
BULLETTRAIN_DIR_FG=039
BULLETTRAIN_DIR_CONTEXT_SHOW=false  # show username & hostname

BULLETTRAIN_GIT_BG=236  # dark grey
BULLETTRAIN_GIT_FG=230  # creme

BULLETTRAIN_GIT_CLEAN="%F{230}   %F{000}"
BULLETTRAIN_GIT_DIRTY="%F{230}  %F{000}" 
BULLETTRAIN_GIT_MODIFIED="%F{230} %F{000}"
BULLETTRAIN_GIT_ADDED="%F{230} %F{000}"
BULLETTRAIN_GIT_DELETED="%F{230} %F{000}"
# BULLETTRAIN_GIT_RENAMED=
BULLETTRAIN_GIT_UNTRACKED="%F{230} %\ %F{000}"
BULLETTRAIN_GIT_AHEAD="%F{230} %\ %\ "
BULLETTRAIN_GIT_BEHIND="%F{039} %\ %\ "
BULLETTRAIN_GIT_DIVERGED="%F{039} %\ %\ "
BULLETTRAIN_GIT_UNMERGED="%F{039} %\ %\ "
BULLETTRAIN_GIT_COLORIZE_DIRTY=true
BULLETTRAIN_GIT_COLORIZE_DIRTY_BG_COLOR=197  # dark grey
BULLETTRAIN_GIT_COLORIZE_DIRTY_FG_COLOR=230  # black
```

**Comment by dualmoon** [comment]
I have this as well. My zshrc is pretty standard but when i change the antigen line to set bullet train as my theme I find that it just sort of.... doesn't. Attached is a screenshot from Windows using wsl-terminal but the same behavior is present on my arch linux laptop.

![zsh-bullet-train](https://user-images.githubusercontent.com/183767/29805791-68a1b5f6-8c51-11e7-86fa-348a5002a5c5.PNG)
